pkg/action/validate.go

```diff
@@ -21,12 +21,22 @@ import (
 
 	"github.com/pkg/errors"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
 	"k8s.io/cli-runtime/pkg/resource"
 
 	"helm.sh/helm/v3/pkg/kube"
 )
 
-func existingResourceConflict(resources kube.ResourceList) error {
+var accessor = meta.NewAccessor()
+
+const (
+	helmReleaseNameAnnotation      = "meta.helm.sh/release-name"
+	helmReleaseNamespaceAnnotation = "meta.helm.sh/release-namespace"
```

Comment:
Interesting thought... How do we want to handle global objects like ClusterRoles? I know they're kind of an edge case, but what is the expected behaviour if they have these annotations? As it stands, it looks like you could add an annotation like

```
meta.helm.sh/release-name: foo
meta.helm.sh/release-namespace: foo
```

To an existing ClusterRole, call `helm install --namespace foo foo` with a chart that installs this ClusterRole and it'll adopt the ClusterRole. Is that intentional behaviour here? If not, how do we want to handle these resources?

Reply:
This is intentional behavior. Please correct me if I'm wrong, but my understanding is that Helm requires every release to be associated with a namespace, even if that release only acts on cluster scoped resources.

Reply:
Yep! Seems like it's in alignment here. So these annotations relate to the name/namespace this object is "owned" by. :+1: